File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    # method1
    # answer = []
    # for i,b in enumerate(flag):
    #     if b:
    #         for j in range(arr[i]*2):
    #             answer.append(arr[i])
    #     else:
    #         for j in range(arr[i]):
    #             answer.pop()
    # method2
    answer=[]
    for i,b in zip(arr,flag):
        if b:
            answer+=[i]*i*2
        else:
            answer=answer[:-i]
    return answer

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// arr_len은 배열 arr의 길이입니다.
// flag_len은 배열 flag의 길이입니다.
int* solution(int arr[], size_t arr_len, bool flag[], size_t flag_len) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* r=(int*)malloc(sizeof(int)*(2000));
    int idx=0;
    for(int i=0;i<flag_len;i++) {
        if(flag[i]) {
            for(int j=0;j<arr[i]*2;j++)
                r[idx++]=arr[i];
        } else {
            idx-=arr[i];
        }
    }
    r=(int*)realloc(r,sizeof(int)*idx);
    return r;
}